hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/TransitRegionStateProcedure.java

```diff
@@ -188,6 +196,27 @@ protected boolean waitInitialized(MasterProcedureEnv env) {
     return am.waitMetaLoaded(this) || am.waitMetaAssigned(this, getRegion());
   }
 
+  private void checkAndWaitForOriginalServer(MasterProcedureEnv env, ServerName lastHost)
+    throws ProcedureSuspendedException {
+    ServerManager serverManager = env.getMasterServices().getServerManager();
+    ServerName newNameForServer = serverManager.findServerWithSameHostnamePortWithLock(lastHost);
+    boolean isOnline = serverManager.createDestinationServersList().contains(newNameForServer);
+
+    if (!isOnline && retries < env.getAssignmentManager().getForceRegionRetainmentRetries()) {
+      retries++;
+      LOG.info("Suspending the TRSP PID={} because {} is true and previous host {} "
+        + "for region is not yet online.", this.getProcId(), FORCE_REGION_RETAINMENT, lastHost);
+      setTimeout(env.getAssignmentManager().getForceRegionRetainmentWait());
+      setState(ProcedureProtos.ProcedureState.WAITING_TIMEOUT);
+      throw new ProcedureSuspendedException();
+    }
+    LOG.info(
```

Comment:
Need to reset retries to 0 here. And better name it waitForOriginalServerRetries.

Reply:
Switched to use RetryCounter and exponential backoff.